netbench-orchestrator/src/orchestrator.rs

```diff
@@ -21,10 +28,267 @@ pub enum RunMode {
+    run_mode: RunMode,
+) -> OrchResult<()> {
+    let iam_client = aws_sdk_iam::Client::new(aws_config);
+    let s3_client = aws_sdk_s3::Client::new(aws_config);
+    let ec2_client = aws_sdk_ec2::Client::new(aws_config);
+    let ssm_client = aws_sdk_ssm::Client::new(aws_config);
+
+    upload_run_parameters_to_s3(&s3_client, config, &unique_id).await?;
+
+    // Setup instances
+    let infra = ec2_utils::LaunchPlan::create(&ec2_client, &iam_client, &ssm_client, config)
+        .await?
+        .launch(&ec2_client, &unique_id)
+        .await?;
+
+    update_dashboard_with_instances(&s3_client, config, &infra, &unique_id).await?;
+
+    run_netbench(
+        run_mode,
+        config,
+        &infra,
+        &ssm_client,
+        &s3_client,
+        &unique_id,
+    )
+    .await?;
+
+    // Cleanup
+    infra
+        .cleanup(&ec2_client)
+        .await
+        .map_err(|err| eprintln!("Failed to cleanup all resources. {err} {:?}", infra))
+        .unwrap();
+
+    Ok(())
+}
+
+async fn upload_run_parameters_to_s3(
+    s3_client: &aws_sdk_s3::Client,
+    config: &OrchestratorConfig,
+    unique_id: &str,
 ) -> OrchResult<()> {
+    let scenario_file = ByteStream::from_path(config.netbench_scenario_filepath())
+        .await
+        .map_err(|err| OrchError::Init {
+            dbg: err.to_string(),
+        })?;
+
+    s3_utils::upload_object(
+        s3_client,
+        config.cdk_config.netbench_runner_public_s3_bucket(),
+        scenario_file,
+        &format!("{unique_id}/{}", config.netbench_scenario_filename()),
+    )
+    .await
+    .unwrap();
+
+    // upload the index.html dashboard file
+    dashboard::upload_index_html(s3_client, unique_id, config).await?;
+
+    Ok(())
+}
+
+async fn update_dashboard_with_instances(
+    s3_client: &aws_sdk_s3::Client,
+    config: &OrchestratorConfig,
+    infra: &InfraDetail,
+    unique_id: &str,
+) -> OrchResult<()> {
+    dashboard::update_instance_running(
+        s3_client,
+        infra,
+        unique_id,
+        config,
+        ec2_utils::EndpointType::Server,
+    )
+    .await?;
+
+    dashboard::update_instance_running(
+        s3_client,
+        infra,
+        unique_id,
+        config,
+        ec2_utils::EndpointType::Client,
+    )
+    .await?;
+
+    Ok(())
+}
+
+async fn run_netbench(
+    run_mode: RunMode,
+    config: &OrchestratorConfig,
+    infra: &InfraDetail,
+    ssm_client: &aws_sdk_ssm::Client,
+    s3_client: &aws_sdk_s3::Client,
+    unique_id: &str,
+) -> OrchResult<()> {
+    if matches!(run_mode, RunMode::Full) {
+        let server_drivers = vec![
+            ssm_utils::s2n_quic_dc_driver::dc_quic_server_driver(unique_id, config),
+            ssm_utils::tcp_driver_crates::tcp_server_driver(),
+            ssm_utils::s2n_quic_driver_crates::s2n_quic_server_driver(),
+            ssm_utils::s2n_tls_driver::s2n_tls_server_driver(),
+            // ssm_utils::native_tls_driver::native_tls_server_driver(),
```

Comment:
why is this commented out? Can it be removed?

Reply:
I wanted to list all available driver. That way a new user can go in there and comment out/in the ones they want.

Reply:
Actually this might deserve a comment/issue.

The tcp driver had issues that I never fully figured out (the client would never finish and cause the netbench process to hang). We decided to continue since TCP was the least interesting of the protocols.

Reply:
https://github.com/aws/s2n-netbench/issues/37